classes/handler/APIHandler.inc.php

```diff
@@ -38,6 +38,17 @@ public function __construct() {
 		parent::__construct();
 		import('lib.pkp.classes.security.authorization.internal.ApiAuthorizationMiddleware');
 		import('lib.pkp.classes.security.authorization.internal.ApiTokenDecodingMiddleware');
+
+		$parts = null;
+		$pathInfoEnabled = Config::getVar('general', 'disable_path_info') ? false : true;
+		$query = parse_url($_SERVER['REQUEST_URI'],PHP_URL_QUERY);
+		parse_str($query, $parts);
+		if (!$pathInfoEnabled && in_array('endpoint', array_keys($parts)) && !empty($parts['endpoint'])) {
+			$endpoint = trim($parts['endpoint'], '/');
+			$requestUri = str_replace('/index.php', "/index.php/{$endpoint}", $_SERVER['REQUEST_URI']);
```

Comment:
What about installs that rewrite to exclude `index.php` from URLs? (See the `restful_urls` setting in `config.inc.php`)

Reply:
I was unaware of such config option. Let me dig into this.

Reply:
@asmecher : this patch does not break installs with `restful_urls` enabled.  `ui-library` always prepend ajax request with `/index.php` whether `restfu_urls` is enabled or not. (cf. https://github.com/pkp/ui-library/blob/master/src/mixins/global.js#L93-L98)

Reply:
@kaschioudi, it looks like that's true for `ui-library` (@NateWr, might this need fixing?) but I don't think it is for URLs generated within OJS. Have you tested with the restful URLs option? In that case, the `index.php` should never be seen.

Reply:
@asmecher Yes, it looks like that could be changed in the API. It doesn't seem to break anything. I run with `restful_urls` enabled, but my API requests to `/index.php/publicknowledge/api/v1/_submissions` work fine.  So `index.php` isn't exposed unless someone looks into the network requests in the browser. Should I file an issue for it?

Reply:
Yes, please do.